### ISSUE TITLE
no link named "go-ipfs_v0.4.15_android-arm64.tar.gz"

### ISSUE THREAD

**Issue by NatoBoram** [ipfs/go-ipfs#5061]
#### Version information:

```bash
ipfs-update -version
ipfs-update version 1.5.3
```

#### Type:

Can't install IPFS trough `ipfs-update` on Android arm64.

#### Description:

```Bash
neofetch --off
u0_a167@localhost
-----------------
OS: Android P aarch64
Host: google Pixel
Kernel: 3.18.100-gdd686e4
Uptime: secs
Packages: 48
Shell: login /data/data/com.termux/files/usr/bin/bash:
CPU: (4) @ 1.593GHz
Memory: 2417MiB / 3762MiB

go get -u -fix github.com/ipfs/ipfs-update
ipfs-update install latest

fetching go-ipfs version v0.4.15
ERROR: fetching resource: 404 Not Found
install failed, reverting changes...
ERROR: failed to get ipfs binary: 404 Not Found: ipfs resolve -r /ipns/dist.ipfs.io/go-ipfs/v0.4.15/go-ipfs_v0.4.15_android-arm64.tar.gz: no link named "go-ipfs_v0.4.15_android-arm64.tar.gz" under QmZtXmnZa6YxhT7WuZph6SL17nWeTNMAyaWKxqWQozPp2V
```

**Comment by NatoBoram** [comment]
I don't know what's happening with GitHub's code blocks right now, I can see my actual comment flashing before my eyes for a second before it gets heavily truncated for unknown reasons.

**Comment by Stebalien** [comment]
There's an android app but I'm not sure how well it works or how up-to-date it is: https://github.com/ligi/IPFSDroid

TL; DR: we don't currently distribute binaries for android. You're welcome to try compiling ipfs yourself but I can't guarantee it'll work.

**Comment by NatoBoram** [comment]
This app isn't stable and crashes when not focused. I have also tried to build it in https://github.com/ipfs/go-ipfs/issues/5063, but that didn't work. The [Linux arm](https://dist.ipfs.io/#go-ipfs) build works, maybe you can add a link so `ipfs-update` gets that one?

**Comment by Stebalien** [comment]
Interesting. Actually, you can probably get this to work by installing the prebuilt [linux-arm ipfs-update](https://dist.ipfs.io/ipfs-update/v1.5.2/ipfs-update_v1.5.2_linux-arm.tar.gz).

The problem here is that ipfs-update always fetches an ipfs build with a matching GOOS/GOARCH (android/arm in this case). I'd rather not add a special-case override as I'm not sure how well the linux build will work on android in general. For example, it looks like running a linux build on android may mis-detect a lack of float support on some phones.

**Comment by NatoBoram** [comment]
That could do it. I actually built IPFS-Update with Termux because it's so easy with Go.

Maybe the Linux-arm build is bound to have errors on Android arm64, but my Firefox using IPFS Companion was able to detect Termux' IPFS node, use it, and I even uploaded [this video](https://ipfs.io/ipfs/QmbvEgddHqEefusYMJQ2c417mwtYE19jquAwpkXJpiHKez) with it.

Instead of an override, maybe you can just copy the linux build in the android slot, so IPFS-Update can at least fetch *something*, even if it's broken on some systems?

**Comment by ligi** [comment]
@NatoBoram can you reproduce the crash of IPFSDroid when it is not focused? If so can you please report it here: https://github.com/ligi/IPFSDroid/issues ?

**Comment by Stebalien** [comment]
> Instead of an override, maybe you can just copy the linux build in the android slot, so IPFS-Update can at least fetch something, even if it's broken on some systems?

I'd rather have nothing by default than a partially broken distribution by default.

**Comment by Stebalien** [comment]
I'm going to close this as there's nothing we can really do about this for now. However, feel free to continue the discussion (we just have too many open issues that we can't really do anything about...).

We can reopen it if something changes.
